src/libstd/sys/unix/rwlock.rs

```diff
@@ -9,16 +9,22 @@
 // except according to those terms.
 
 use libc;
-use cell::UnsafeCell;
+use cell::{Cell, UnsafeCell};
 
-pub struct RWLock { inner: UnsafeCell<libc::pthread_rwlock_t> }
+pub struct RWLock {
+    inner: UnsafeCell<libc::pthread_rwlock_t>,
+    write_locked: Cell<bool>,
```

Comment:
This is technically accessed concurrently so could this switch to being an `UnsafeCell` as well?